components/AlertsDashboard.vue

```diff
@@ -1973,8 +2090,454 @@ const calculateLineStringCentroid = (coordinates: number[][]) => {
+  boundingBoxMode.value = false;
+};
+
+/**
+ * Toggles bounding box mode for selecting features within a drawn area
+ */
+const toggleBoundingBoxMode = () => {
+  boundingBoxMode.value = !boundingBoxMode.value;
+  multiSelectMode.value = false;
+
+  if (boundingBoxMode.value) {
+    if (map.value) {
+      map.value.dragRotate.disable();
+    }
+    setupCustomBoundingBox();
+  } else {
+    if (map.value) {
+      map.value.dragRotate.enable();
+    }
+    removeBoundingBoxHandlers();
+  }
+};
+
+/**
+ * Sets up custom bounding box selection (Ctrl/Cmd + drag)
+ * Creates a visual selection box and selects features within it
+ */
+const setupCustomBoundingBox = () => {
+  if (!map.value) return;
+
+  const canvas = map.value.getCanvasContainer();
+  let start: mapboxgl.Point | null = null;
+  let current: mapboxgl.Point | null = null;
+  let box: HTMLElement | null = null;
+
+  // Add CSS for the selection box
+  if (!document.getElementById("bounding-box-styles")) {
+    const style = document.createElement("style");
+    style.id = "bounding-box-styles";
+    style.textContent = `
+      .boxdraw {
+        background: rgba(56, 135, 190, 0.1);
+        border: 2px solid #3887be;
+        position: absolute;
+        top: 0;
+        left: 0;
+        width: 0;
+        height: 0;
+        pointer-events: none;
+        z-index: 1000;
+      }
+    `;
+    document.head.appendChild(style);
+  }
+
+  /**
+   * Returns the xy coordinates of the mouse position relative to the canvas
+   */
+  const mousePos = (e: MouseEvent) => {
+    const rect = canvas.getBoundingClientRect();
+    return new mapboxgl.Point(
+      e.clientX - rect.left - canvas.clientLeft,
+      e.clientY - rect.top - canvas.clientTop,
+    );
+  };
+
+  /**
+   * Handles mouse down event to start bounding box selection
+   */
+  const mouseDown = (e: MouseEvent) => {
+    if (!((e.ctrlKey || e.metaKey) && e.button === 0)) return;
+
+    map.value.dragPan.disable();
+
+    document.addEventListener("mousemove", onMouseMove);
+    document.addEventListener("mouseup", onMouseUp);
+    document.addEventListener("keydown", onKeyDown);
+
+    start = mousePos(e);
+  };
+
+  /**
+   * Handles mouse move to draw the selection box
+   */
+  const onMouseMove = (e: MouseEvent) => {
+    if (!start) return;
+
+    current = mousePos(e);
+
+    if (!box) {
+      box = document.createElement("div");
+      box.classList.add("boxdraw");
+      canvas.appendChild(box);
+    }
+
+    const minX = Math.min(start.x, current.x);
+    const maxX = Math.max(start.x, current.x);
+    const minY = Math.min(start.y, current.y);
+    const maxY = Math.max(start.y, current.y);
+
+    const pos = `translate(${minX}px, ${minY}px)`;
+    box.style.transform = pos;
+    box.style.width = maxX - minX + "px";
+    box.style.height = maxY - minY + "px";
+  };
+
+  /**
+   * Handles mouse up to finish selection
+   */
+  const onMouseUp = (e: MouseEvent) => {
```

Comment:
Functions like `onMouseUp`, `finish` are actually very specific to bounding box selection for incidents, not generic to all of the AlertsDashboard.

At the very least I'd suggest making the function names a bit more precise: e.g. `onMouseUpDuringBoundingBox` `finishBoundingBox`

However a better recommendation may be to try to split a lot of this out to 1 or more new component(s).  This file is almost 3000 lines long and really has a runaway scope.  I think you could fairly easily extract this PR's new stuff to one or more new components, and perhaps another time later we come back and also look at if we should break apart any of the code that was here prior.